### PR TITLE
Add swift support

### LINKED ISSUE #139: Swift
**Is your feature request related to a problem? Please describe.**
I'd like to be able to use Zed to work on my Swift projects.

**Describe the solution you'd like**
[SourceKit-LSP](https://github.com/apple/sourcekit-lsp) could be used.

**Describe alternatives you've considered**
N/A?

**Screenshots**
N/A

### LINKED ISSUE #5278: Swift support
**Is your feature request related to a problem? Please describe.**
I'd like to be able to use Zed to work on my Swift projects.

**Describe the solution you'd like**
[SourceKit-LSP](https://github.com/apple/sourcekit-lsp) could be used.

**Describe alternatives you've considered**
N/A?

**Screenshots**
N/A

### PR DESCRIPTION
**Pull request by ceopaludetto**
This PR adds the Swift tree-sitter grammar copied from [nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter/tree/master/queries/swift). It also adds the [sourcekit-lsp](https://github.com/apple/sourcekit-lsp).

This PR resolves zed-industries/extensions#139

Release Notes:
- Added Swift support (#5278).

### CODE REVIEW

**Comment by ceopaludetto** [comment]
@cla-bot check

File: crates/zed/src/languages/swift/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+
+(function_declaration
+  "init" @constructor)
+
+(throws) @keyword
+
+(where_keyword) @keyword
+
+(parameter
+  external_name: (simple_identifier) @variable.parameter)
```
Comment (mikayla-maki) [inline crates/zed/src/languages/swift/highlights.scm:50]: We don't have a `@variable.parameter`, you can see what we do have in [syntax.rs](https://github.com/zed-industries/zed/blob/main/crates/theme/src/styles/syntax.rs#L16-L68)

**Review by mikayla-maki — COMMENTED** [review]
Thank you for the PR! I think it would be nice to see integration with Zed's outline view, via an `outline.scm` file. Also, I don't think we can add any Swift PR until we've discussed what to do about: https://github.com/alex-pinkus/tree-sitter-swift/issues/240

**Comment by ceopaludetto** [comment]
> Thank you for the PR! I think it would be nice to see integration with Zed's outline view, via an `outline.scm` file. Also, I don't think we can add any Swift PR until we've discussed what to do about: [alex-pinkus/tree-sitter-swift#240](https://github.com/alex-pinkus/tree-sitter-swift/issues/240)

Yeah, the highlight slowness is a big problem as discussed in #5278. But can Swift be added behind a feature flag(something like "unstable swift support") until the issue is resolved?

**Comment by marcusziade** [comment]
The slowdown is only one initial load thing according to the author. If that's the case, I'd use this and deal with it. It's still a value add. Can be iterated upon later.

File: crates/zed/src/languages/swift/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+  ">="
+  "++"
+  "--"
+  "&"
+  "~"
+  "%="
+  "!="
+  "!=="
+  "=="
+  "==="
```
Comment (marcusziade) [inline crates/zed/src/languages/swift/highlights.scm:284]: Is this needed?

**Comment by Dog** [comment]
If there isn't support for feature flags (I haven't checked) then perhaps just have it be off by default in settings. Then the documentation can give the caveats and warning. That way people can use the language without needing the jump through building from source, and the impression isn't given its fully supported.

**Comment by marcusziade** [comment]
We should determine which one is better
* [#6832] 

Cc @matyunya

**Comment by ConradIrwin** [comment]
Closing this for now. We have a Swift extension for tree-sitter, and language server support for extensions will land soon.

**Comment by marcusziade** [comment]
> Closing this for now. We have a Swift extension for tree-sitter, and language server support for extensions will land soon.

Can I follow the LSP dev somewhere?

**Comment by ConradIrwin** [comment]
@maxbrunsfeld is working on it (often in https://zed.dev/channel/language-servers-4159). A prototype landed here: https://github.com/zed-industries/zed/pull/9085 if you want to take a look.

**Pull request closed without merging**